File: readme_examples/js/4_object.hasOwn.js
/**
 * Create an instance of Object, with no prototype.
 */
ryo = Object.create(null, {foo: {value: "foo"}});

/**
 * Create a second object, with the "ryo" object as
 * its prototype.
 */
ryo2 = Object.create(ryo, {bar: {value: "bar"}});


/**
 * Returns false
 */
console.log(Object.hasOwn(ryo2, "foo"));

/**
 * Returns true
 */
console.log(Object.hasOwn(ryo2, "bar"));

File: readme_examples/rb/4_object.hasOwn.rb
# frozen_string_literal: true

require_relative "setup"
require "ryo"
require "ryo/core_ext/object"

##
# Create an instance of Object, with no prototype.
ryo = Object.create(nil, {foo: "foo"})

##
# Create a second object, with the "ryo" object as
# its prototype.
ryo2 = Object.create(ryo, {bar: "bar"})

##
# Returns false
p Ryo.property?(ryo2, "foo")

##
# Returns true
p Ryo.property?(ryo2, "bar")
